exporter/honeycomb: Collector crashes if Node.ServiceInfo not set on TraceData.

Zipkin and Jaeger clients support sending trace data without populating `Node.ServiceInfo` on `TraceData` instances. This will currently cause the collector to crash if the honeycomb exporter is configured as part of a pipeline exporting those traces. We should check that the field is not nil before trying to set it as an attribute.

[A copy of the panic and stack trace](https://honeycombpollinators.slack.com/archives/CNQ943Q75/p1585430836019800?thread_ts=1585342471.007700&cid=CNQ943Q75) is available in the "discuss-opentelemetry-beta" channel of the "Honeycomb Pollinators" Slack team, reproduced here:
<details>
   <summary>Go panic</summary>

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x1e5f472]

goroutine 110 [running]:
github.com/open-telemetry/opentelemetry-collector-contrib/exporter/honeycombexporter.(*HoneycombExporter).pushTraceData(0xc000406468, 0x2c0fc60, 0xc0002c7920, 0x0, 0x0, 0xc0002c75f0, 0x6, 0x6, 0x27620e3, 0x6, ...)
	/home/circleci/project/exporter/honeycombexporter/honeycomb.go:63 +0x182
github.com/open-telemetry/opentelemetry-collector/exporter/exporterhelper.traceDataPusher.withObservability.func1(0x2c0fc60, 0xc0002c7920, 0x0, 0x0, 0xc0002c75f0, 0x6, 0x6, 0x27620e3, 0x6, 0x5e7fc033, ...)
	/go/pkg/mod/github.com/open-telemetry/opentelemetry-collector@v0.2.8/exporter/exporterhelper/tracehelper.go:98 +0xc9
github.com/open-telemetry/opentelemetry-collector/exporter/exporterhelper.(*traceExporter).ConsumeTraceData(0xc0001a3f60, 0x2c0fc60, 0xc0002c76e0, 0x0, 0x0, 0xc0002c75f0, 0x6, 0x6, 0x27620e3, 0x6, ...)
	/go/pkg/mod/github.com/open-telemetry/opentelemetry-collector@v0.2.8/exporter/exporterhelper/tracehelper.go:48 +0xa7
github.com/open-telemetry/opentelemetry-collector/processor/queuedprocessor.(*queuedSpanProcessor).processItemFromQueue(0xc00008bb00, 0xc0001ab560)
	/go/pkg/mod/github.com/open-telemetry/opentelemetry-collector@v0.2.8/processor/queuedprocessor/queued_processor.go:159 +0xf8
github.com/open-telemetry/opentelemetry-collector/processor/queuedprocessor.(*queuedSpanProcessor).Start.func1(0x20d8300, 0xc0001ab560)
	/go/pkg/mod/github.com/open-telemetry/opentelemetry-collector@v0.2.8/processor/queuedprocessor/queued_processor.go:92 +0x45
github.com/jaegertracing/jaeger/pkg/queue.(*BoundedQueue).StartConsumers.func1(0xc0004106e0, 0xc0000c43c0)
	/go/pkg/mod/github.com/jaegertracing/jaeger@v1.17.0/pkg/queue/bounded_queue.go:77 +0xb9
created by github.com/jaegertracing/jaeger/pkg/queue.(*BoundedQueue).StartConsumers
	/go/pkg/mod/github.com/jaegertracing/jaeger@v1.17.0/pkg/queue/bounded_queue.go:68 +0xc0
```

</details>

--- a/exporter/honeycombexporter/honeycomb.go
+++ b/exporter/honeycombexporter/honeycomb.go
@@ -59,11 +59,16 @@ func (e *HoneycombExporter) pushTraceData(ctx context.Context, td consumerdata.T
 	for _, span := range td.Spans {
 		sd, err := honeycomb.OCProtoSpanToOTelSpanData(span)
 		if err == nil {
-			serviceName := core.Key("service_name")
-			sd.Attributes = append(sd.Attributes, serviceName.String(td.Node.ServiceInfo.Name))
-			if !sd.ParentSpanID.IsValid() || sd.HasRemoteParent {
+			if td.Node != nil && td.Node.ServiceInfo != nil {
+				serviceName := core.Key("service_name")
 				sd.Attributes = append(sd.Attributes,
-					convertNodeAttributes(td.Node.Attributes)...)
+					serviceName.String(td.Node.ServiceInfo.Name))
+			}
+			if !sd.ParentSpanID.IsValid() || sd.HasRemoteParent {
+				if td.Node != nil && td.Node.Attributes != nil {
+					sd.Attributes = append(sd.Attributes,
+						convertNodeAttributes(td.Node.Attributes)...)
+				}
 			}
 			e.exporter.ExportSpan(ctx, sd)
 			goodSpans++

--- a/exporter/honeycombexporter/honeycomb_test.go
+++ b/exporter/honeycombexporter/honeycomb_test.go
@@ -36,35 +36,37 @@ type honeycombData struct {
 	Data map[string]interface{} `json:"data"`
 }
 
-func TestExporter(t *testing.T) {
-	var got []honeycombData
-
-	server := httptest.NewServer(http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
+func testingServer(callback func(data []honeycombData)) *httptest.Server {
+	return httptest.NewServer(http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
 		uncompressed, err := zstd.NewReader(req.Body)
 		if err != nil {
-			http.Error(rw, err.Error(), 500)
+			http.Error(rw, err.Error(), http.StatusInternalServerError)
 			return
 		}
 		defer req.Body.Close()
 		b, err := ioutil.ReadAll(uncompressed)
 		if err != nil {
-			http.Error(rw, err.Error(), 500)
+			http.Error(rw, err.Error(), http.StatusInternalServerError)
 			return
 		}
 
 		var data []honeycombData
 		err = json.Unmarshal(b, &data)
 		if err != nil {
-			http.Error(rw, err.Error(), 500)
+			http.Error(rw, err.Error(), http.StatusInternalServerError)
 			return
 		}
-
-		got = append(got, data...)
-
+		callback(data)
 		rw.Write([]byte(`OK`))
 	}))
-	defer server.Close()
+}
 
+func testTraceExporter(td consumerdata.TraceData, t *testing.T) []honeycombData {
+	var got []honeycombData
+	server := testingServer(func(data []honeycombData) {
+		got = append(got, data...)
+	})
+	defer server.Close()
 	cfg := Config{
 		APIKey:  "test",
 		Dataset: "test",
@@ -76,6 +78,15 @@ func TestExporter(t *testing.T) {
 	exporter, err := factory.CreateTraceExporter(logger, &cfg)
 	require.NoError(t, err)
 
+	ctx := context.Background()
+	err = exporter.ConsumeTraceData(ctx, td)
+	require.NoError(t, err)
+	exporter.Shutdown()
+
+	return got
+}
+
+func TestExporter(t *testing.T) {
 	td := consumerdata.TraceData{
 		Node: &commonpb.Node{
 			ServiceInfo: &commonpb.ServiceInfo{Name: "test_service"},
@@ -109,12 +120,7 @@ func TestExporter(t *testing.T) {
 			},
 		},
 	}
-
-	ctx := context.Background()
-	err = exporter.ConsumeTraceData(ctx, td)
-	require.NoError(t, err)
-	exporter.Shutdown()
-
+	got := testTraceExporter(td, t)
 	want := []honeycombData{
 		{
 			Data: map[string]interface{}{
@@ -154,6 +160,41 @@ func TestExporter(t *testing.T) {
 				"trace.span_id":     "0400000000000000",
 				"trace.trace_id":    "01000000-0000-0000-0000-000000000000",
 				"A":                 "B",
+			},
+		},
+	}
+
+	if diff := cmp.Diff(want, got); diff != "" {
+		t.Errorf("otel span: (-want +got):\n%s", diff)
+	}
+}
+
+func TestEmptyNode(t *testing.T) {
+	td := consumerdata.TraceData{
+		Node: nil,
+		Spans: []*tracepb.Span{
+			{
+				TraceId:                 []byte{0x01},
+				SpanId:                  []byte{0x02},
+				Name:                    &tracepb.TruncatableString{Value: "root"},
+				Kind:                    tracepb.Span_SERVER,
+				SameProcessAsParentSpan: &wrappers.BoolValue{Value: true},
+			},
+		},
+	}
+
+	got := testTraceExporter(td, t)
+
+	want := []honeycombData{
+		{
+			Data: map[string]interface{}{
+				"duration_ms":       float64(0),
+				"has_remote_parent": false,
+				"name":              "root",
+				"status.code":       float64(0),
+				"status.message":    "OK",
+				"trace.span_id":     "0200000000000000",
+				"trace.trace_id":    "01000000-0000-0000-0000-000000000000",
 			},
 		},
 	}

